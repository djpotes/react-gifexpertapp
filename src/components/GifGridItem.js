import PropTypes from 'prop-types';
import React from 'react'

export const GifGridItem = ({id, title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn" >
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}