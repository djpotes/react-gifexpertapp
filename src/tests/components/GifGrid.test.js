import '@testing-library/jest-dom'

import {GifGrid} from "../../components/GifGrid"
import {shallow}  from "enzyme"
import {useFetchGifs} from '../../hooks/useFetchGifs'

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas del component <GifGrid />', () =>{

    const category = 'Hola Mundo';
    let wrapper;
    test('Debe de mostrar el component correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs =  [{
            id: 'ABC',
            url: 'https://localhost/casas.jpg',
            title: 'Casa'
        },
        {
            id: '123',
            url: 'https://localhost/casas.jpg',
            title: 'Casa'
        }
        ]
        useFetchGifs.mockReturnValue({data:gifs,loadig:false})
        wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    });
})