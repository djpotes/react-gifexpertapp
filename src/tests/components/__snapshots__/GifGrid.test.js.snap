// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del component <GifGrid /> Debe de mostrar el component correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola Mundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas del component <GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Hola Mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Casa"
      url="https://localhost/casas.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Casa"
      url="https://localhost/casas.jpg"
    />
  </div>
</Fragment>
`;
